policy/filter_test.go

```diff
@@ -0,0 +1,196 @@
+package policy
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	hclog "github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/require"
+)
+
+// TestFilteredSource_MonitorIDs_FilterInput tests that MonitorIDs
+// filters the upstream policy IDs
+func TestFilteredSource_MonitorIDs_FilterInput(t *testing.T) {
+	require := require.New(t)
+
+	monitorCtx, monitorCancel := context.WithCancel(context.Background())
+
+	inputCh := make(chan IDMessage)
+	errCh := make(chan error)
+	testFilter := NewTesterFilter(nil)
+	testSource := NewTestSource(inputCh, errCh)
+
+	source := NewFilteredSource(hclog.NewNullLogger(), testSource, testFilter)
+	outputCh := make(chan IDMessage)
+	outputErrCh := make(chan error)
+	monitorExited := make(chan bool)
+	go func() {
+		source.MonitorIDs(monitorCtx, MonitorIDsReq{
+			ErrCh:    outputErrCh,
+			ResultCh: outputCh,
+		})
+		monitorExited <- true
+	}()
+
+	// send the message from the upstream
+	expected := []PolicyID{
+		"abcde",
+		"a1234",
+		"aaaaa",
+	}
+	unexpected := []PolicyID{
+		"badbad",
+		"zzzzzz",
+		"123456",
+	}
+	go func() {
+		inputCh <- IDMessage{
+			IDs:    append(expected, unexpected...),
+			Source: "test",
+		}
+	}()
+
+	// should not receive a message before BOTH filter and upstream have messaged
+	select {
+	case <-outputCh:
+		require.Fail("received message before filter sent message")
+	case <-time.After(2 * time.Second):
```

Comment:
Maybe reduce these timeouts (or make it configurable on the number of tests running? I'm not even sure if that's possible 😅) 

2s doesn't seem like a lot but it could compound when running the full test suite.

Reply:
The only way I can figure off the top of my head is to use env vars; but that in itself will likely get lost and be tricky/annoying to use in local setups. I'll make a note to keep this in mind in the future.